### PR TITLE
[cli] Add "out-of-band" mode to `vc login`

### PR DESCRIPTION
**Pull request by TooTallNate**
Adds an "out-of-band" login mode to the `vc login` command, for use in headless environments like inside a Docker container or an SSH session. In these situations, spawning a localhost HTTP server won't work, because the HTTP server will not be accessible to the web browser that is doing the authentication.

In "out-of-band" mode, the login URL is simply presented to the user, and they must visit the login URL in their web browser and complete the authentication. The callback URL is a page on front that shows the verification token to the user and instructs them to copy + paste the token back into the CLI, at which point the login can complete.

Docker/SSH sessions are attempted to be automatically detected, and "out-of-band" mode is enabled implicitly. But there is also an `--oob` flag in case that detection logic were to fail (or if you'd like to run OOB mode locally).

This PR also adds shortcut flags for the Git provider login mechanisms, i.e. `vc login --github`.

https://user-images.githubusercontent.com/71256/125129304-1f5f3380-e0b4-11eb-8139-d1109811bd86.mov

### CODE REVIEW

File: packages/cli/src/util/login/oauth.ts [hunk 1]
```diff
@@ -14,14 +15,57 @@ export default async function doOauthLogin(
...
+  if (typeof result === 'number') {
+    return result;
+  }
 
+  if ('verificationToken' in result) {
+    const { output } = client;
+    output.spinner('Verifying authentication token');
+    result = await verify(
+      client,
+      'nate@vercel.com',
```
Comment (zach-jayz) [inline packages/cli/src/util/login/oauth.ts]: leftover placeholder?

**Review by zach-jayz — APPROVED** [review]
LGTM

File: packages/cli/src/util/login/oauth.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/cli/src/util/login/oauth.ts]: Yup. Removed.

**Review by pacocoursey — APPROVED** [review]

File: packages/cli/src/commands/login.ts [hunk 2]
```diff
@@ -41,6 +44,10 @@ const help = () => {
   ${chalk.gray('–')} Log in using a specific team "slug" for SAML Single Sign-On
 
     ${chalk.cyan(`$ ${getPkgName()} login acme`)}
+
+  ${chalk.gray('–')} Log in using GitHub in "out-of-band" mode
```
Comment (styfle) [inline packages/cli/src/commands/login.ts:48]: Is there precedent for this terminology?

How about `--headless` instead?

Comment (TooTallNate) [reply packages/cli/src/commands/login.ts:48]: I copied the terminology from [Twitter](https://developer.twitter.com/en/docs/authentication/oauth-1-0a/pin-based-oauth)'s solution to this same problem.

**Review by styfle — APPROVED** [review]
Looks good!

Don't forget to update the docs https://vercel.com/docs/cli

**Pull request merged**